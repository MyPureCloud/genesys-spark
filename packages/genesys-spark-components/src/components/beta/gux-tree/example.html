<h1>gux-tree-beta</h1>

<gux-tree-beta onchange="notify(event)" oninput="notify(event)">
  <gux-branch value="b-1">
    <gux-icon
      slot="icon"
      icon-name="fa/code-branch-regular"
      decorative
    ></gux-icon>
    <div slot="label">Branch 1</div>

    <gux-leaf value="l-1-1" disabled>
      <gux-icon
        slot="icon"
        icon-name="fa/diamond-regular"
        decorative
      ></gux-icon>
      <div slot="label">Leaf 1-1</div>
    </gux-leaf>

    <gux-leaf value="l-1-2" disabled>
      <gux-icon
        slot="icon"
        icon-name="fa/diamond-regular"
        decorative
      ></gux-icon>
      <div slot="label">Leaf 1-2</div>
    </gux-leaf>
  </gux-branch>

  <gux-branch>
    <gux-icon
      slot="icon"
      icon-name="fa/code-branch-regular"
      decorative
    ></gux-icon>
    <div slot="label">Branch 2</div>

    <gux-branch value="b-2-1" disabled>
      <gux-icon
        slot="icon"
        icon-name="fa/code-branch-regular"
        decorative
      ></gux-icon>
      <div slot="label">Branch 2-1</div>

      <gux-leaf value="l-2-1-1">
        <gux-icon
          slot="icon"
          icon-name="fa/diamond-regular"
          decorative
        ></gux-icon>
        <div slot="label">Leaf 2-1-1</div>
      </gux-leaf>

      <gux-leaf value="l-2-1-2">
        <gux-icon
          slot="icon"
          icon-name="fa/diamond-regular"
          decorative
        ></gux-icon>
        <div slot="label">Leaf 2-1-2</div>
      </gux-leaf>

      <gux-leaf value="l-2-1-3">
        <gux-icon
          slot="icon"
          icon-name="fa/diamond-regular"
          decorative
        ></gux-icon>
        <div slot="label">Leaf 2-1-3</div>
      </gux-leaf>
    </gux-branch>

    <gux-branch value="b-2-2">
      <gux-icon
        slot="icon"
        icon-name="fa/code-branch-regular"
        decorative
      ></gux-icon>
      <div slot="label">Branch 2-2</div>

      <gux-leaf value="l-2-2-1">
        <gux-icon
          slot="icon"
          icon-name="fa/diamond-regular"
          decorative
        ></gux-icon>
        <div slot="label">Leaf 2-2-1</div>
      </gux-leaf>

      <gux-leaf value="l-2-2-2">
        <gux-icon
          slot="icon"
          icon-name="fa/diamond-regular"
          decorative
        ></gux-icon>
        <div slot="label">Leaf 2-2-2</div>
      </gux-leaf>
    </gux-branch>
  </gux-branch>

  <gux-leaf value="l-1">
    <gux-icon slot="icon" icon-name="fa/diamond-regular" decorative></gux-icon>
    <div slot="label">Leaf 1</div>
  </gux-leaf>
</gux-tree-beta>
