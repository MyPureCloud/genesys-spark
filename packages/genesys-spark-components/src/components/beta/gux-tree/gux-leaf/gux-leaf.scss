:host {
  display: block;
}

button {
  min-block-size: 24px;
}

.gux-target {
  display: flex;
  gap: 10px;
  padding: 5px;
  margin: 5px;
  border: 1px solid green;

  &.gux-selected {
    background-color: lightgrey;
  }

  &.gux-disabled {
    opacity: 0.5;
  }
}
