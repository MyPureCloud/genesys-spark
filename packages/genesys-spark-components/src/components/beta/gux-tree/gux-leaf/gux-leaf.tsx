import {
  AttachInternals,
  Component,
  Element,
  h,
  JSX,
  Prop,
  Watch
} from '@stencil/core';

/**
 * @slot label - Required slot for the text
 * @slot icon - Optional slot for an icon
 */

@Component({
  styleUrl: 'gux-leaf.scss',
  tag: 'gux-leaf',
  formAssociated: true,
  shadow: true
})
export class GuxLeaf {
  @Element()
  root: HTMLElement;

  @AttachInternals()
  internals: ElementInternals;

  @Prop()
  value: string;

  @Prop()
  selected: boolean = false;

  @Prop()
  disabled: boolean = false;

  @Prop()
  active: boolean = false;

  @Prop()
  filtered: boolean = false;

  @Watch('selected')
  handleSelectedChange(newValue: boolean) {
    this.internals.ariaSelected = newValue.toString();
    this.root.setAttribute('aria-selected', newValue.toString());
  }

  @Watch('active')
  handleActiveChange(active: boolean) {
    this.root.setAttribute('tabIndex', active ? '0' : '');

    if (active) {
      this.root.focus();
    }
  }

  componentWillLoad() {
    this.internals.role = 'treeitem';
    this.root.setAttribute('role', 'treeitem');
    this.root.setAttribute('tabIndex', this.active ? '0' : '');
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-target': true,
          'gux-disabled': this.disabled,
          'gux-selected': this.selected
        }}
      >
        <div class="gux-icon">
          <slot name="icon"></slot>
        </div>
        <div class="gux-label">
          <slot name="label"></slot>
        </div>
      </div>
    ) as JSX.Element;
  }
}
