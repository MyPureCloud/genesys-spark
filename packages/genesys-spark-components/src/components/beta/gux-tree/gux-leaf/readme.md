# gux-leaf

<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type      | Default     |
| ---------- | ---------- | ----------- | --------- | ----------- |
| `active`   | `active`   |             | `boolean` | `false`     |
| `disabled` | `disabled` |             | `boolean` | `false`     |
| `filtered` | `filtered` |             | `boolean` | `false`     |
| `selected` | `selected` |             | `boolean` | `false`     |
| `value`    | `value`    |             | `string`  | `undefined` |


## Slots

| Slot      | Description                |
| --------- | -------------------------- |
| `"icon"`  | Optional slot for an icon  |
| `"label"` | Required slot for the text |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
