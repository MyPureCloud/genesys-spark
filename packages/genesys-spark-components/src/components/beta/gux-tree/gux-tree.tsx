import {
  AttachInternals,
  Component,
  Element,
  h,
  JSX,
  Prop,
  Listen,
  Watch,
  forceUpdate
} from '@stencil/core';

import {
  handleTreeNodeSpecificEvent,
  setFirstNodeActive,
  setInitialActiveNode,
  setLastNodeActive,
  setNextNodeActive,
  setPreviousNodeActive,
  setSelectedNode,
  setParentNodeActive,
  setChildNodeActive,
  setSelectedNodeByValue
} from './gux-tree.service';

import { trackComponent } from '@utils/tracking/usage';
import simulateNativeEvent from '@utils/dom/simulate-native-event';

/**
 * @slot label - slot for label
 * @slot search - slot for search
 * @slot default - gux-tree-view-branch or gux-tree-view-leaf elements
 */

@Component({
  styleUrl: 'gux-tree.scss',
  tag: 'gux-tree-beta',
  formAssociated: true,
  shadow: true
})
export class GuxTreeBeta {
  @Element()
  root: HTMLGuxTreeBetaElement;

  @AttachInternals()
  internals: ElementInternals;

  @Prop({ mutable: true })
  value: string;

  @Watch('value')
  handleValueChange(): void {
    simulateNativeEvent(this.root, 'input');
    simulateNativeEvent(this.root, 'change');
  }

  @Listen('click')
  onClick(event: MouseEvent): void {
    handleTreeNodeSpecificEvent({
      event,
      onLeaf: leaf => {
        setSelectedNode(this.root, leaf);
      },
      onBranch: branch => {
        branch.expanded = !branch.expanded;
        setSelectedNode(this.root, branch);
      }
    });
  }

  @Listen('keydown')
  handleKeydown(event: KeyboardEvent): void {
    switch (event.key) {
      case 'ArrowDown':
        event.preventDefault();
        setNextNodeActive(this.root);
        break;
      case 'ArrowUp':
        event.preventDefault();
        setPreviousNodeActive(this.root);
        break;
      case 'Home':
        event.preventDefault();
        setFirstNodeActive(this.root);
        break;
      case 'End':
        event.preventDefault();
        setLastNodeActive(this.root);
        break;
      case 'ArrowLeft':
      case 'ArrowRight':
      case 'Enter':
        handleTreeNodeSpecificEvent({
          event,
          onLeaf: leaf => {
            switch (event.key) {
              case 'ArrowLeft':
                event.preventDefault();
                setParentNodeActive(this.root);
                break;
              case 'Enter':
                event.preventDefault();
                setSelectedNode(this.root, leaf);
                break;
            }
          },
          onBranch: branch => {
            switch (event.key) {
              case 'ArrowRight':
                event.preventDefault();
                if (!branch.expanded) {
                  branch.expanded = true;
                } else {
                  setChildNodeActive(this.root);
                }
                break;
              case 'ArrowLeft':
                event.preventDefault();
                if (branch.expanded) {
                  branch.expanded = false;
                } else {
                  setParentNodeActive(this.root);
                }
                break;
              case 'Enter':
                event.preventDefault();
                setSelectedNode(this.root, branch);
                break;
            }
          }
        });
        break;
      default:
        return;
    }
  }

  componentWillLoad() {
    trackComponent(this.root);
    this.internals.role = 'tree';
    this.root.setAttribute('role', 'tree');

    setSelectedNodeByValue(this.root, this.value);
    setInitialActiveNode(this.root);
  }

  private onDefaultSlotchange(): void {
    forceUpdate(this.root);
  }

  render(): JSX.Element {
    return (
      <slot onSlotchange={() => this.onDefaultSlotchange()} />
    ) as JSX.Element;
  }
}
