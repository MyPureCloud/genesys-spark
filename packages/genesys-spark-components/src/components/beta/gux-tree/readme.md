# gux-tree



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default     |
| -------- | --------- | ----------- | -------- | ----------- |
| `value`  | `value`   |             | `string` | `undefined` |


## Slots

| Slot        | Description                                         |
| ----------- | --------------------------------------------------- |
| `"default"` | gux-tree-view-branch or gux-tree-view-leaf elements |
| `"label"`   | slot for label                                      |
| `"search"`  | slot for search                                     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
