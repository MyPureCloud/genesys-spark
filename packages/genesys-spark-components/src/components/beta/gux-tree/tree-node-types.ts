/**
 * This is the interface tree items are required to adhere to. It's mostly a
 * collection of states for the individual elements that needs to be queried
 * or set by the listbox.
 */
export interface TreeNodeElement extends HTMLElement {
  value?: string;
  disabled: boolean;
  filtered: boolean;
  active: boolean;
  selected: boolean;
  expanded?: boolean;
}
