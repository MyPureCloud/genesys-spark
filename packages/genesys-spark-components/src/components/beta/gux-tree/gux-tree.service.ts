import { whenEventIsFrom } from '@utils/dom/when-event-is-from';

import { TreeNodeElement } from './tree-node-types';

// Utility functions for node type checking
function isNode(item: Element): boolean {
  const nodeTypes = ['GUX-LEAF', 'GUX-BRANCH'];
  return nodeTypes.includes(item.tagName);
}

function isBranchNode(item: Element): boolean {
  return item.tagName === 'GUX-BRANCH';
}

function isOpenBranchNode(item: Element): boolean {
  if (item.tagName === 'GUX-BRANCH') {
    return (item as HTMLGuxBranchElement).expanded;
  }

  return false;
}

function hasActiveNode(tree: HTMLGuxTreeBetaElement): boolean {
  return Boolean(getActiveNode(tree));
}

// Core tree traversal functions
function getAllTreeNodes(tree: HTMLGuxTreeBetaElement): TreeNodeElement[] {
  const nodes: TreeNodeElement[] = [];

  function getNodes(element: Element): void {
    for (const child of Array.from(element.children)) {
      if (isNode(child)) {
        nodes.push(child as TreeNodeElement);
      }

      if (isBranchNode(child)) {
        getNodes(child);
      }
    }
  }

  getNodes(tree);

  return nodes;
}

function getTreeNodes(tree: HTMLGuxTreeBetaElement): TreeNodeElement[] {
  const nodes: TreeNodeElement[] = [];

  function getNodes(element: Element): void {
    for (const child of Array.from(element.children)) {
      if (isNode(child)) {
        nodes.push(child as TreeNodeElement);
      }

      if (isOpenBranchNode(child)) {
        getNodes(child);
      }
    }
  }

  getNodes(tree);

  return nodes;
}

function getAvailableTreeNodes(
  tree: HTMLGuxTreeBetaElement
): TreeNodeElement[] {
  const availableTreeNodes = getTreeNodes(tree).filter(node => {
    return !node.disabled && !node.filtered;
  });

  return availableTreeNodes;
}

// Node getter functions
function getActiveNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  return getTreeNodes(tree).find(node => node.active);
}

function getFirstSelectedNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  return getTreeNodes(tree).find(
    node => node.selected && !node.disabled && !node.filtered
  );
}

function getFirstNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  return getAvailableTreeNodes(tree)[0];
}

function getLastNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  const availableNodes = getAvailableTreeNodes(tree);
  return availableNodes[availableNodes.length - 1];
}

function getNextNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  if (hasActiveNode(tree) && getActiveNode(tree) !== getLastNode(tree)) {
    const availableTreeNodes = getAvailableTreeNodes(tree);
    const activeNode = getActiveNode(tree);
    const activeNodeIndex = availableTreeNodes.indexOf(activeNode);

    return availableTreeNodes[activeNodeIndex + 1];
  }

  return getFirstNode(tree);
}

function getPreviousNode(tree: HTMLGuxTreeBetaElement): TreeNodeElement {
  if (hasActiveNode(tree)) {
    const availableNodesList = getAvailableTreeNodes(tree);
    const activeNodeIndex = availableNodesList.indexOf(getActiveNode(tree));

    if (getActiveNode(tree) === getFirstNode(tree)) {
      return getLastNode(tree);
    }

    return availableNodesList[activeNodeIndex - 1];
  }

  return getFirstNode(tree);
}

function getNodeByValue(
  tree: HTMLGuxTreeBetaElement,
  value: string
): TreeNodeElement {
  return getAllTreeNodes(tree).find(node => node.value === value);
}

// Core setter functions
function setActiveNode(
  tree: HTMLGuxTreeBetaElement,
  element: TreeNodeElement
): void {
  if (element) {
    getTreeNodes(tree).forEach(node => {
      const active = !node.disabled && !node.filtered && node === element;
      node.active = active;

      if (active) {
        tree.setAttribute('aria-activedescendant', node.id);
      }
    });
    element.scrollIntoView({ block: 'nearest' });
  }
}

function expandAllParents(element: TreeNodeElement): void {
  let parent = element.parentElement.closest('gux-branch');

  while (parent) {
    parent.expanded = true;
    parent = parent.parentElement.closest('gux-branch');
  }
}

// Public API - Active node management
export function setInitialActiveNode(tree: HTMLGuxTreeBetaElement) {
  setActiveNode(tree, getFirstSelectedNode(tree) || getFirstNode(tree));
}

export function setFirstNodeActive(tree: HTMLGuxTreeBetaElement): void {
  setActiveNode(tree, getFirstNode(tree));
}

export function setLastNodeActive(tree: HTMLGuxTreeBetaElement): void {
  setActiveNode(tree, getLastNode(tree));
}

export function setNextNodeActive(tree: HTMLGuxTreeBetaElement): void {
  setActiveNode(tree, getNextNode(tree));
}

export function setPreviousNodeActive(tree: HTMLGuxTreeBetaElement): void {
  setActiveNode(tree, getPreviousNode(tree));
}

export function setParentNodeActive(tree: HTMLGuxTreeBetaElement): void {
  const activeNode = getActiveNode(tree);
  const parentNode = activeNode.parentElement.closest('gux-branch');

  setActiveNode(tree, parentNode);
}

export function setChildNodeActive(tree: HTMLGuxTreeBetaElement): void {
  const branch = getActiveNode(tree) as HTMLGuxBranchElement;
  const availableNodes = getAvailableTreeNodes(branch);

  if (availableNodes[0]) {
    setActiveNode(tree, availableNodes[0]);
  }
}

// Public API - Selection management
export function setSelectedNode(
  tree: HTMLGuxTreeBetaElement,
  element: TreeNodeElement
): void {
  if (element?.value) {
    getTreeNodes(tree).forEach(node => {
      node.selected = node === element;
    });

    tree.value = element.value;
    element.scrollIntoView({ block: 'nearest' });
  }
}

export function setSelectedNodeByValue(
  tree: HTMLGuxTreeBetaElement,
  value: string
): void {
  const element = getNodeByValue(tree, value);
  if (element) {
    expandAllParents(element);
    setSelectedNode(tree, element);
  }
}

// Public API - Event handling
export function handleTreeNodeSpecificEvent({
  event,
  onLeaf,
  onBranch
}: {
  event: Event | PointerEvent | KeyboardEvent;
  onLeaf: (leaf: HTMLGuxLeafElement) => void;
  onBranch: (branch: HTMLGuxBranchElement) => void;
}): void {
  let eventFromLeaf = false;
  whenEventIsFrom('gux-leaf', event, (leaf: HTMLGuxLeafElement) => {
    if (leaf.value) {
      onLeaf(leaf);
      eventFromLeaf = true;
    }
  });
  if (!eventFromLeaf) {
    whenEventIsFrom('gux-branch', event, (branch: HTMLGuxBranchElement) => {
      onBranch(branch);
    });
  }
}
