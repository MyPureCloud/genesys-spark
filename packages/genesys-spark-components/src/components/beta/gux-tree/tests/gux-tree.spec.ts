import { checkRenders } from '@test/specTestUtils';

import { GuxTreeBeta } from '../gux-tree';
import { GuxBranch } from '../gux-branch/gux-branch';
import { GuxLeaf } from '../gux-leaf/gux-leaf';

import { renderConfigs } from './gux-tree.common';

const components = [GuxTreeBeta, GuxBranch, GuxLeaf];

describe('gux-time', () => {
  describe('#render', () => {
    checkRenders(renderConfigs, components);
  });
});
