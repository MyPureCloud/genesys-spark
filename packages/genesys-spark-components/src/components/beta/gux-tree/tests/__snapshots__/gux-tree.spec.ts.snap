// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`gux-time #render should render as expected 1`] = `
<gux-tree-beta role="tree" aria-activedescendant="">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <gux-branch tabindex="0" role="treeitem" aria-expanded="false">
    <template shadowrootmode="open">
      <div class="gux-container">
        <div class="gux-target">
          <div class="gux-branch-indicator">
            <gux-icon class="gux-arrow-icon" decorative="" icon-name="fa/caret-right-solid" size="small">
            </gux-icon>
          </div>
          <div class="gux-icon">
            <slot name="icon">
            </slot>
          </div>
          <div class="gux-label">
            <slot name="label">
            </slot>
          </div>
        </div>
        <div role="group" class="gux-treeitems">
          <slot>
          </slot>
        </div>
      </div>
    </template>
    <div slot="label">
      Projects
    </div>
    <gux-leaf role="treeitem" tabindex="">
      <template shadowrootmode="open">
        <div class="gux-target">
          <div class="gux-icon">
            <slot name="icon">
            </slot>
          </div>
          <div class="gux-label">
            <slot name="label">
            </slot>
          </div>
        </div>
      </template>
      <div slot="label">
        Project-1
      </div>
    </gux-leaf>
    <gux-leaf role="treeitem" tabindex="">
      <template shadowrootmode="open">
        <div class="gux-target">
          <div class="gux-icon">
            <slot name="icon">
            </slot>
          </div>
          <div class="gux-label">
            <slot name="label">
            </slot>
          </div>
        </div>
      </template>
      <div slot="label">
        Project-2
      </div>
    </gux-leaf>
  </gux-branch>
  <gux-branch role="treeitem" tabindex="" aria-expanded="false">
    <template shadowrootmode="open">
      <div class="gux-container">
        <div class="gux-target">
          <div class="gux-branch-indicator">
            <gux-icon class="gux-arrow-icon" decorative="" icon-name="fa/caret-right-solid" size="small">
            </gux-icon>
          </div>
          <div class="gux-icon">
            <slot name="icon">
            </slot>
          </div>
          <div class="gux-label">
            <slot name="label">
            </slot>
          </div>
        </div>
        <div role="group" class="gux-treeitems">
          <slot>
          </slot>
        </div>
      </div>
    </template>
    <div slot="label">
      Reports
    </div>
    <gux-branch role="treeitem" tabindex="" aria-expanded="false">
      <template shadowrootmode="open">
        <div class="gux-container">
          <div class="gux-target">
            <div class="gux-branch-indicator">
              <gux-icon class="gux-arrow-icon" decorative="" icon-name="fa/caret-right-solid" size="small">
              </gux-icon>
            </div>
            <div class="gux-icon">
              <slot name="icon">
              </slot>
            </div>
            <div class="gux-label">
              <slot name="label">
              </slot>
            </div>
          </div>
          <div role="group" class="gux-treeitems">
            <slot>
            </slot>
          </div>
        </div>
      </template>
      <div slot="label">
        2025
      </div>
      <gux-leaf role="treeitem" tabindex="">
        <template shadowrootmode="open">
          <div class="gux-target">
            <div class="gux-icon">
              <slot name="icon">
              </slot>
            </div>
            <div class="gux-label">
              <slot name="label">
              </slot>
            </div>
          </div>
        </template>
        <div slot="label">
          doc 1
        </div>
      </gux-leaf>
      <gux-leaf role="treeitem" tabindex="">
        <template shadowrootmode="open">
          <div class="gux-target">
            <div class="gux-icon">
              <slot name="icon">
              </slot>
            </div>
            <div class="gux-label">
              <slot name="label">
              </slot>
            </div>
          </div>
        </template>
        <div slot="label">
          doc 2
        </div>
      </gux-leaf>
      <gux-leaf role="treeitem" tabindex="">
        <template shadowrootmode="open">
          <div class="gux-target">
            <div class="gux-icon">
              <slot name="icon">
              </slot>
            </div>
            <div class="gux-label">
              <slot name="label">
              </slot>
            </div>
          </div>
        </template>
        <div slot="label">
          doc 3
        </div>
      </gux-leaf>
    </gux-branch>
    <gux-branch role="treeitem" tabindex="" aria-expanded="false">
      <template shadowrootmode="open">
        <div class="gux-container">
          <div class="gux-target">
            <div class="gux-branch-indicator">
              <gux-icon class="gux-arrow-icon" decorative="" icon-name="fa/caret-right-solid" size="small">
              </gux-icon>
            </div>
            <div class="gux-icon">
              <slot name="icon">
              </slot>
            </div>
            <div class="gux-label">
              <slot name="label">
              </slot>
            </div>
          </div>
          <div role="group" class="gux-treeitems">
            <slot>
            </slot>
          </div>
        </div>
      </template>
      <div slot="label">
        2024
      </div>
    </gux-branch>
  </gux-branch>
  <gux-leaf role="treeitem" tabindex="">
    <template shadowrootmode="open">
      <div class="gux-target">
        <div class="gux-icon">
          <slot name="icon">
          </slot>
        </div>
        <div class="gux-label">
          <slot name="label">
          </slot>
        </div>
      </div>
    </template>
    <div slot="label">
      Untitled Documents
    </div>
  </gux-leaf>
</gux-tree-beta>
`;
